### PR TITLE
findById() fetches the record which is soft deleted in DB

### LINKED ISSUE #11: findById() fetches the record which is soft deleted in DB
**Describe the bug**
findById returns the record successfully, even if the record is soft deleted in DB.

**To Reproduce**
1. Call method findById()
2. it will return the record even if the record is soft deleted.

**Expected behavior**
If the record is soft-deleted, it should not return the record.

### PR DESCRIPTION
**Pull request by nareshkharola123**
## Description

fixed issue of findById returns the record successfully, even if the record is soft deleted in DB.
also new dependency added: @loopback/rest

Fixes #11 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested ?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [X] Performed a self-review of my own code
- [X] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [X] Code conforms with the style guide
- [ ] API Documentation in code was updated

### CODE REVIEW

File: src/repositories/soft-crud.repository.base.ts [hunk 1]
```diff
@@ -137,12 +141,24 @@ export abstract class SoftCrudRepository<
...
+          id: id,
+        } as Condition<T>,
+      };
     }
 
-    // Now call super
-    return super.findById(id, filter, options);
+    //As parent method findById have filter: FilterExcludingWhere<T>
+    //so we need add check here.
+    const checkSoftDeleted = await super.findOne(filter, options);
```
Comment (samarpan-b) [inline src/repositories/soft-crud.repository.base.ts:154]: rename it to entityToRemove

Comment (nareshkharola123) [reply src/repositories/soft-crud.repository.base.ts]: Done

**Comment by samarpan-b** [comment]
test case ?

**Comment by nareshkharola123** [comment]
> test case ?

working on it..

**Comment by nareshkharola123** [comment]
Hi, As we discussed, we are not going to do unit-testing.

**Pull request merged**